File: codegen/results/v002/qwen3.5_35b-a3b-coding-nvfp4_think/20260414_103016/go/005_unit_test_writer/solution.go
package main

import "fmt"

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func assertApproxEqual(expected, got, tolerance float64, description string) {
	if abs(expected-got) <= tolerance {
		fmt.Printf("PASS: %s\n", description)
	} else {
		fmt.Printf("FAIL: %s (expected: %.2f, got: %.2f)\n", description, expected, got)
	}
}

func abs(x float64) float64 {
	if x < 0 {
		return -x
	}
	return x
}

func main() {
	testCases := []struct {
		price       float64
		quantity    int
		expected    float64
		description string
	}{
		{price: 100.00, quantity: 9, expected: 900.00, description: "No discount (quantity < 10)"},
		{price: 100.00, quantity: 10, expected: 900.00, description: "10% discount (quantity = 10)"},
		{price: 100.00, quantity: 49, expected: 4410.00, description: "10% discount (quantity = 49)"},
		{price: 100.00, quantity: 50, expected: 3750.00, description: "25% discount (quantity = 50)"},
	}

	for _, tc := range testCases {
		result := calculateDiscount(tc.price, tc.quantity)
		assertApproxEqual(tc.expected, result, 0.01, tc.description)
	}
}

File: codegen/results/v002/qwen3.5_35b-a3b-coding-nvfp4_think/20260414_103016/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

function runTest(
    description: string,
    price: number,
    quantity: number,
    expected: number
): void {
    const actual = calculateDiscount(price, quantity);
    const tolerance = 0.0001;
    const passed = Math.abs(actual - expected) < tolerance;

    if (passed) {
        console.log(`PASS: ${description}`);
    } else {
        console.log(`FAIL: ${description} (expected: ${expected}, got: ${actual})`);
    }
}

// Test cases covering boundary conditions
runTest("quantity = 9 (no discount)", 100, 9, 900);
runTest("quantity = 10 (10% discount)", 100, 10, 900);
runTest("quantity = 49 (10% discount)", 100, 49, 4410);
runTest("quantity = 50 (25% discount)", 100, 50, 3750);

// Additional test cases
runTest("quantity = 1 (no discount)", 100, 1, 100);
runTest("quantity = 100 (25% discount)", 100, 100, 7500);

// Program exit
process.exit(0);